src/StepAdaptationController/include/WalkingControllers/StepAdaptationController/StepAdaptationController.hpp

```diff
@@ -0,0 +1,272 @@
+         * @return true/false in case of success/failure.
+         */
+        bool solve();
+
+        /**
+         * Get the solver solution
+         * @return the entire solution of the solver
+         */
+        const iDynTree::VectorDynSize& getSolution() const;
+
+        /**
+         * Initialize the method
+         * @param config yarp searchable configuration variable.
+         * @return true/false in case of success/failure
+         */
+        bool initialize(const yarp::os::Searchable& config);
+
+        /**
+         * Solve the Optimization problem. If the QPSolver is not set It will be initialized.
+         * @param isLeft is true if left foot is the swing foot
+         * @return true/false in case of success/failure.
+         */
+        bool solve(bool isLeft);
+
+        /**
+         * Reset the controller
+         */
+        void reset();
+
+        /**
+         * Set the nominal next step position and yaw angle
+         * @param nominalZmpPosition Nominal next step position(with a constant offset)
+         * @param angle yaw angle of the swing foot at landing moment.
+         * @return true/false in case of success/failure.
+         */
+        void setNominalNextStepPosition(const iDynTree::Vector2& nominalZmpPosition, const double& angle);
+
+        /**
+         * Set the varibales related to the timing of a step
+         * @param omega Nominal next step position(with a constant offset).
+         * @param currentTime current time of walking.
+         * @param nextImpactTime Next impact time
+         * @param nextDoubleSupportDuration Double support duration of the next step.
+         */
+        void setTimings(const double & omega, const double & currentTime, const double& nextImpactTime,
+                        const double &nextDoubleSupportDuration);
+
+        /**
+         * Set the nominal DCM offset.
+         * @param nominalDcmOffset Nominal DCM offset.
+         * @return true/false in case of success/failure.
+         */
+        void setNominalDcmOffset(const iDynTree::Vector2& nominalDcmOffset);
+
+        /**
+         * Set the current ZMP(or with a constant offset the next step) Position.
+         * @param currentZmpPosition Current position of the zmp.
+         * @return true/false in case of success/failure.
+         */
+        void setCurrentZmpPosition(const iDynTree::Vector2& currentZmpPosition);
+
+        /**
+         * Set the current Dcm Position.
+         * @param currentDcmPosition Current position of the DCM.
+         * @return true/false in case of success/failure.
+         */
+        void setCurrentDcmPosition(const iDynTree::Vector2& currentDcmPosition);
+
+        /**
+         * Get the output of the controller.
+         * @param controllerOutput is the vector containing the output the controller.
+         * @return true/false in case of success/failure.
+         */
+        bool getControllerOutput(iDynTree::Vector3& controllerOutput);
```

Comment:
Is this needed outside? If so, it is not clear what ``controllerOutput`` is.

Reply:
We can receive the control output by their specific name!
However by this function we can access to all of them by a vector!

Reply:
But the vectorization is unknown and easily confused. I would suggest simplifying the interface and remove it. What do you think?

Reply:
I think you are right. I removed that.